### PR TITLE
feat(axis): add visibility to tick style

### LINKED ISSUE #330: Remove tick lines
**Describe the bug**
The tick lines that are visible between the tick label and the grid adds lots of noise to the tick labels. Instead we should be relying on the grid to mark out the values across the whole chart so users can actually follow a line to the correct value.

**Expected behavior**
I tried removing them by setting `theme.tickLineStyle.strokeWidth = 0`. While this does make them disappear, there is still a lot of extra space between the tick label and the axis line leftover from where the tick line should be.

image

**Either,** 
A. just get rid of the idea of tick lines
B. make sure we can fully remove theme via the theme

**Version (please complete the following information):**
 - Elastic Charts: 9.0.4

### PR DESCRIPTION
**Pull request by nickofthyme**
## Summary

Add visibility property to `tickLineStyle` in `Theme`.

BREAKING CHANGE: `theme.axes.tickLineStyle.visible` is now required (default base is `false`)

closes #330

Screen Recording 2019-09-12 at 01 20 PM

### Checklist
- [x] Proper documentation or storybook story was added for features that require explanation or tutorials
- [x] Unit tests were updated or added to match the most common scenarios
- [x] Each commit follows the [convention](https://github.com/elastic/elastic-charts/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by markov00** [comment]
Can we have the default visibility to true, and change the default to false when we are going to implement a better default style?

**Comment by nickofthyme** [comment]
@markov00 I set the default to false as that is the style EUI is suggesting. Are you saying to set the default to true and then convert to false later?

**Comment by markov00** [comment]
Yes, I would like to have a single PR that adjust the default theme to a better one. For example in this case hide the ticks but also reduce the tick size to 0 for example. Together with some better defaults of style for the axis colors, grid style, line annotations etc

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@elastic/charts",
   "description": "Elastic-Charts data visualization library",
-  "version": "12.0.2",
+  "version": "12.0.0",
```
Comment (markov00) [inline package.json]: we cannot go back

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,17 +1,3 @@
...
-
-* **reactive_chart:** fix order of instantiation of onBruchEnd callback ([#376](https://github.com/elastic/elastic-charts/issues/376)) ([527d68d](https://github.com/elastic/elastic-charts/commit/527d68d)), closes [#360](https://github.com/elastic/elastic-charts/issues/360)
-
-## [12.0.1](https://github.com/elastic/elastic-charts/compare/v12.0.0...v12.0.1) (2019-09-12)
-
-
-### Bug Fixes
-
-* **theme:** fix grid position check ([#373](https://github.com/elastic/elastic-charts/issues/373)) ([af4805f](https://github.com/elastic/elastic-charts/commit/af4805f)), closes [#372](https://github.com/elastic/elastic-charts/issues/372)
-
```
Comment (markov00) [inline CHANGELOG.md]: Why does changes?

File: package.json (hunk 1 above)
Comment (nickofthyme) [reply package.json]: Sorry I didn't see that, must have been a bad merge. Fix in 6eda1fa

File: CHANGELOG.md (hunk 2 above)
Comment (nickofthyme) [reply CHANGELOG.md]: Fix in 6eda1fa

**Comment by markov00** [comment]
jenkins, test this please

**Pull request merged**